js/react_native/ios/OnnxruntimeModuleTest/OnnxruntimeModuleTest.mm

```diff
@@ -16,6 +16,71 @@ @implementation OnnxruntimeModuleTest
 - (void)testOnnxruntimeModule {
   NSBundle *bundle = [NSBundle bundleForClass:[OnnxruntimeModuleTest class]];
   NSString *dataPath = [bundle pathForResource:@"test_types_float" ofType:@"ort"];
+  NSString *sessionKey = @"";
+
+  OnnxruntimeModule *onnxruntimeModule = [OnnxruntimeModule new];
+
+  // test loadModelFromBuffer()
+  {
+    NSMutableDictionary *options = [NSMutableDictionary dictionary];
+    NSData *fileData = [NSData dataWithContentsOfFile: dataPath];
+    NSDictionary *resultMap = [onnxruntimeModule loadModelFromBuffer:fileData options:options];
+
+    sessionKey = resultMap[@"key"];
+    NSArray *inputNames = resultMap[@"inputNames"];
+    XCTAssertEqual([inputNames count], 1);
+    XCTAssertEqualObjects(inputNames[0], @"input");
+    NSArray *outputNames = resultMap[@"outputNames"];
+    XCTAssertEqual([outputNames count], 1);
+    XCTAssertEqualObjects(outputNames[0], @"output");
+  }
+
+  // test run()
+  {
+    NSMutableDictionary *inputTensorMap = [NSMutableDictionary dictionary];
+
+    // dims
+    NSArray *dims = @[ [NSNumber numberWithLong:1], [NSNumber numberWithLong:5] ];
+    inputTensorMap[@"dims"] = dims;
+
+    // type
+    inputTensorMap[@"type"] = JsTensorTypeFloat;
+
+    // data
+    std::array<float_t, 5> outValues{std::numeric_limits<float_t>::min(), 1.0f, -2.0f, 3.0f,
+                                     std::numeric_limits<float_t>::max()};
+
+    const NSInteger byteBufferSize = outValues.size() * sizeof(float_t);
+    unsigned char *byteBuffer = static_cast<unsigned char *>(malloc(byteBufferSize));
+    NSData *byteBufferRef = [NSData dataWithBytesNoCopy:byteBuffer length:byteBufferSize];
+    float *floatPtr = (float *)[byteBufferRef bytes];
+    for (NSUInteger i = 0; i < outValues.size(); ++i) {
+      *floatPtr++ = outValues[i];
+    }
+    floatPtr = (float *)[byteBufferRef bytes];
+
+    NSString *dataEncoded = [byteBufferRef base64EncodedStringWithOptions:0];
+    inputTensorMap[@"data"] = dataEncoded;
+
+    NSMutableDictionary *inputDataMap = [NSMutableDictionary dictionary];
+    inputDataMap[@"input"] = inputTensorMap;
+
+    NSMutableDictionary *options = [NSMutableDictionary dictionary];
+
+    NSMutableArray *output = [NSMutableArray array];
+    [output addObject:@"output"];
+
+    NSDictionary *resultMap = [onnxruntimeModule run:sessionKey input:inputDataMap output:output options:options];
+
+    XCTAssertTrue([[resultMap objectForKey:@"output"] isEqualToDictionary:inputTensorMap]);
```

Comment:
Once the session is created, is the rest of the code common? If so could we put it in a helper function and just pass in the sessionKey?

Reply:
Yes, I should put the common code in a helper function.

but another thing I just noticed is, it seems like when running unit tests for both paths, it can fail.

i.e. if I put individual test for either loadmodel+run/loadmodelfrombuffer+run, the test will pass.

but putting both tests here, OnnxruntimemoduleTest will fail. (will fail at execution stage, the session can be successfully initialized. ) also checked the session key, looks about right, i.e. (= 0 and 1).

wondering is there any hidden reason that may cause this?

Reply:
@fs-eire might know. I'm not aware of any. 

Does it fail if you explicitly close/cleanup the session in the test (not sure how to do that though). 

Are the tests running sequentially or concurrently? If not sure, what happens if both tests are in one function so they're forced to be sequential?

Reply:
currently there is no session cleanup. should not be the reason.

is the test failure still exist?

Reply:
ok, got it. Previously the tests from the React Native CI pipeline got all passed for this branch after the update.